Files/Helpers/FileListCache/FileListCacheController.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.Extensions.Caching.Memory;
+using System.Threading.Tasks;
+
+namespace Files.Helpers.FileListCache
+{
+    internal class FileListCacheController : IFileListCache
+    {
+        private static FileListCacheController instance;
+        public static FileListCacheController GetInstance()
+        {
+            return instance ??= new FileListCacheController();
+        }
+
+        private readonly IFileListCache persistentAdapter;
+
+        private FileListCacheController()
+        {
+            persistentAdapter = new PersistentSQLiteCacheAdapter();
+        }
+
+        private readonly IMemoryCache filesCache = new MemoryCache(new MemoryCacheOptions
+        {
+            SizeLimit = 1_000_000
+        });
+
+        public Task SaveFileListToCache(string path, CacheEntry cacheEntry)
+        {
+            if (!App.AppSettings.UseFileListCache) return Task.CompletedTask;
```

Comment:
@tsvietOK Should this be changedd to `ExperimentalViewModel`?

Reply:
`ExperimentalViewModel` is not available until it is launched.